### ISSUE TITLE
GGally package issue

### ISSUE THREAD

**Issue by Simonvln** [ggobi/ggally#449]
I am really struggling with GGally package, the main issue is after installing the GGally package based on **CRAN** i am not able to load it to start the interpretation  !

it shows like the following : 

> library(GGally)
**Error: package or namespace load failed for ‘GGally’ in readRDS(mapfile):
 unknown input format**

However when I tired to install from **GitHub** that did not work

> > install_github("ggobi/ggally")
Downloading GitHub repo ggobi/ggally@HEAD
✔  checking for file ‘/private/var/folders/57/yr8d__cd0dq2htw8jbl5w3pm0000gn/T/Rtmpzk2B0D/remotes26d2457a3de/ggobi-ggally-53c1f65/DESCRIPTION’ (428ms)
─  preparing ‘GGally’:
✔  checking DESCRIPTION meta-information ...
─  installing the package to process help pages
         -----------------------------------
─  installing *source* package ‘GGally’ ...
   ** using staged installation
   ** R
   ** data
   *** moving datasets to lazyload DB
   ** inst
   ** byte-compile and prepare package for lazy loading
   Error in readRDS(mapfile) : unknown input format
   Calls: <Anonymous> ... loadNamespace -> lazyLoad -> lazyLoadDBexec -> readRDS
   Execution halted
   ERROR: lazy loading failed for package ‘GGally’
─  removing ‘/private/var/folders/57/yr8d__cd0dq2htw8jbl5w3pm0000gn/T/RtmpAlicSn/Rinst520429fe2e4/GGally’
         -----------------------------------
   **ERROR: package installation failed
Error: Failed to install 'GGally' from GitHub:
  ! System command 'R' failed**

---

inability to load GGally package

**Comment by schloerke** [comment]
I do not believe this is related to GGally directly, but package installation in general.

I do see others having the same issue with the error message `readRDS(mapfile) : unknown input format`, but no one posted a successful fix. 😞 

------------------------------

Random check: Have you tried restarting your machine?

Possible, but drastic, solution: Go nuclear and delete your R library and reinstall packages as needed. Not fun, but it gives you a fresh start.

To get around the headache of trying to library packages that won't exist anymore, I have this method in my `.Rprofile` to help me load packages that don't exist upon request.

```r
if(interactive()) {

  library <- function(...) {
    # Only work if called by the user
    if (!identical(parent.frame(), globalenv())) {
      return(base::library(...))
    }

    pkg <- as.character(substitute(...))

    # No package given
    if (length(pkg) == 0) {
      return(base::library())
    }

    # Try to find info about the package
    pkgDesc <- base::suppressWarnings(utils::packageDescription(pkg))
    if (identical(pkgDesc, NA)) {
      message("Installing ", pkg, "...")
      utils::install.packages(pkg)
    }

    invisible(base::library(pkg, character.only = TRUE))
  }
}
```

This allows me to execute commands like `library(GGally)` and it will try to install `{GGally}` if `{GGally}` is not found on my machine. Then the package will be loaded with `base::library()`.

**Comment by schloerke** [comment]
Closing as not GGally related
